database/002_initial_schema.sql

```diff
@@ -1,5 +1,470 @@
+);
+
+--OWUser
+CREATE TABLE
+IF NOT EXISTS public.onlineuser(
+    OID SERIAL PRIMARY KEY,
+    JWT varchar NOT NULL,
+    companies int[]
+);
+
+--Event
 CREATE TABLE
-IF NOT EXISTS public.keys
-(accesstoken text UNIQUE);
+IF NOT EXISTS public.event(
+    EID SERIAL PRIMARY KEY,
+    statusCode statues NOT NULL
+    );
+
+--Offer
+CREATE TABLE
+IF NOT EXISTS public.offer(
+    OFID SERIAL PRIMARY KEY,
+    statusCode statues NOT NULL,
+    data jsonb NOT NULL,
+    EID INTEGER REFERENCES event(EID) NOT NULL,
+    CID INTEGER REFERENCES company(CID) NOT NULL,
+    OID INTEGER REFERENCES onlineuser(OID) NOT NULL
+);
+
+--Wish
+CREATE TABLE
+IF NOT EXISTS public.wish(
+    WID SERIAL PRIMARY KEY,
+    statusCode statues NOT NULL,
+    data JSONB NOT NULL,
+    EID INTEGER REFERENCES event(EID) NOT NULL,
+    CUID INTEGER REFERENCES companyuser(CUID),
+    CID INTEGER REFERENCES company(CID) NOT NULL
+);
+--Contracts
+CREATE TABLE
+IF NOT EXISTS public.contracts(
+    contractID SERIAL PRIMARY KEY,
+    acceptDate bigint NOT NULL,
+    expirationDate bigint NOT NULL,
+    price bigint NOT NULL,
+    billingAdresse text NOT NULL,
+    statusCode statues NOT NULL,
+    path text NOT NULL,
+    orgNum bigint REFERENCES company(orgNum) NOT NULL,
+    EID INTEGER REFERENCES event(EID) NOT NULL,
+    CUID INTEGER REFERENCES companyuser(CUID),
+    CID INTEGER REFERENCES company(CID) NOT NULL,
+    OID INTEGER REFERENCES onlineuser(OID) NOT NULL
+);
+--Ads
+CREATE TABLE
+IF NOT EXISTS ad(
+    CID INTEGER REFERENCES company(CID) NOT NULL
+    ) INHERITS (event)
+;
+-- Queue
+CREATE TABLE
+IF NOT EXISTS public.queue(
+    QID SERIAL PRIMARY KEY,
+    statusCode INTEGER NOT NULL,
+    --Epoch Time Format
+    enteredQueue bigint
+        NOT NULL DEFAULT
+        (date_part
+        ('epoch'::text, now
+        ()) *
+        (1000)::double precision),
+    CID INTEGER REFERENCES company(CID) NOT NULL,
+    EID INTEGER REFERENCES event(EID) NOT NULL
+);
+-- Signatures
+CREATE TABLE
+IF NOT EXISTS public.signatures(
+    SID SERIAL PRIMARY KEY,
+    path text,
+    OID INTEGER REFERENCES onlineuser(OID) NOT NULL
+);
+
+
+--bookingevent
+CREATE TABLE
+IF NOT EXISTS bookingevent(
+    data jsonb NOT NULL
+    ) INHERITS (event)
+;
+
+--itex
+CREATE TABLE
+IF NOT EXISTS itex(
+    data jsonb NOT NULL
+    ) INHERITS (event)
+;
+
+
+--Enable row security
+
+ALTER TABLE onlineuser ENABLE ROW LEVEL SECURITY;
+ALTER TABLE company ENABLE ROW LEVEL SECURITY;
+ALTER TABLE companyuser ENABLE ROW LEVEL SECURITY;
+ALTER TABLE event ENABLE ROW LEVEL SECURITY;
+ALTER TABLE offer ENABLE ROW LEVEL SECURITY;
+ALTER TABLE wish ENABLE ROW LEVEL SECURITY;
+ALTER TABLE contracts ENABLE ROW LEVEL SECURITY;
+ALTER TABLE ad ENABLE ROW LEVEL SECURITY;
+ALTER TABLE queue ENABLE ROW LEVEL SECURITY;
+ALTER TABLE signatures ENABLE ROW LEVEL SECURITY;
+ALTER TABLE bookingevent ENABLE ROW LEVEL SECURITY;
+ALTER TABLE itex ENABLE ROW LEVEL SECURITY;
```

Comment:
I'm not able to find a good way of looping over these tables. If you know of a good way to accomplish this i would appreciate the help. If not, i think i'll need to let this stay as is for a while.

Reply:
A little time since i have done much fancy things with PG.

My initial thought would be to fetch all tables in the DB and then loop though all om them altering the table. If it isn't possible to select table names in PG maybe a cursor is needed.

Something like this: https://stackoverflow.com/questions/15266345/loop-on-tables-with-pl-pgsql-in-postgres-9-0

Maybe a bit overkill, anyway a nice thing to know of